collectors/python.d.plugin/spigotmc/spigotmc.chart.py

```diff
@@ -91,27 +105,28 @@ def _get_data(self):
         data = {}
         try:
             raw = self.console.command('tps')
-            # The above command returns a string that looks like this:
-            # '§6TPS from last 1m, 5m, 15m: §a19.99, §a19.99, §a19.99\n'
-            # The values we care about are the three numbers after the :
-            tmp = raw.split(':')[1].split(',')
-            data['tps1'] = float(tmp[0].lstrip(u' §a*')) * PRECISION
-            data['tps5'] = float(tmp[1].lstrip(u' §a*')) * PRECISION
-            data['tps15'] = float(tmp[2].lstrip(u' §a*').rstrip()) * PRECISION
+            match = _TPS_REGEX.match(raw)
+            if match:
+                data['tps1'] = int(match.group(1)) * PRECISION
+                data['tps5'] = int(match.group(2)) * PRECISION
+                data['tps15'] = int(match.group(3)) * PRECISION
+            else:
+                self.error('Unable to process TPS values.')
         except mcrcon.MCRconException:
             self.error('Unable to fetch TPS values.')
         except socket.error:
             self.error('Connection is dead.')
             self.alive = False
             return None
-        except (TypeError, LookupError):
+        except (TypeError, LookupError, IndexError):
```

Comment:
if regex matches we always have 3 x `([0-9]{1,2}.[0-9]{1,2})`

do we need this?

> except (TypeError, LookupError, IndexError):

Reply:
Probably not.